api/contacts/contact.controller.js

```diff
@@ -0,0 +1,86 @@
+const {
+  listContacts,
+  getContactById,
+  removeContact,
+  updateContact,
+  addContact,
+} = require("../../contacts");
+import Joi from "joi";
+import { createControllerProxy } from "../helpers/controllerProxy";
+
+class ContactController {
+  createContact(req, res, next) {
+    try {
+      const newContact = {
+        ...req.body,
+      };
+      addContact(newContact);
+
+      return res.status(201).json(newContact);
+    } catch (err) {
+      next(next);
+    }
+  }
+
+  validateNewContact(req, res, next) {
+    const contactRules = Joi.object({
+      name: Joi.string().required(),
+      email: Joi.string().required(),
+      phone: Joi.string().required(),
+    });
+
+    const validateResult = Joi.validate(req.body, contactRules);
+    if (validateResult.error) {
+      return res
+        .status(400)
+        .json(`MissingFields:${validateResult.error.details[0].message}`);
+    }
+    next();
+  }
+
+  validateUpdatedContact(req, res, next) {
+    if (req.body.constructor === Object && Object.keys(req.body).length === 0) {
```

Comment:
this is overengeneering

should be: 

if (!Object.keys(req.body).length) {

Reply:
This has been fixed and committed. please check